stream-helpers/tracker.ts

```diff
@@ -0,0 +1,74 @@
+import { type Operation, resource, type Stream } from "effection";
+import { createSetSignal, is } from "./signals.ts";
+
+export interface Tracker extends Operation<void> {
+  /**
+   * Returns a stream helper that doesn't modify the items passing through the stream,
+   * but will capture a reference to the item. Call the `markOne` or `markMany` methods
+   * with the item to indicate that it has exited the stream.
+   */
+  passthrough(): <T>(stream: Stream<T, never>) => Stream<T, never>;
```

Comment:
Let's make this fully general with `TClose` so any stream will work, not just those that are infinite.